### PR TITLE
feat: Allow changing languages

### PR DESCRIPTION
**Pull request by Ushie**

### CODE REVIEW

**Comment by validcube** [comment]
This will result in conflict for all PR targeting dev.

I suggest that we merged all PR before merging this.

File: assets/i18n/strings.i18n.json [hunk 1]
```diff
@@ -6,8 +6,8 @@
     "updateButton": "Update",
     "enabledLabel": "Enabled",
     "disabledLabel": "Disabled",
-    "installed":"Installed: {version}",
-    "suggested":"Suggested: {version}",
+    "installed":"Installed: ${version}",
+    "suggested":"Suggested: ${version}",
```
Comment (validcube) [inline assets/i18n/strings.i18n.json:10]: Missing space

```suggestion
    "installed": "Installed: ${version}",
    "suggested": "Suggested: ${version}",
```

File: assets/i18n/strings.i18n.json [hunk 2]
```diff
@@ -123,7 +123,7 @@
 
         "doneButton": "Done",
 
-        "default": "Default",
+        "defaultText": "defaultText",
```
Comment (validcube) [inline assets/i18n/strings.i18n.json:126]: defaultText... *is this intended?*

```suggestion
        "defaultText": "Default",
```

File: pubspec.yaml [hunk 3]
```diff
@@ -101,3 +102,5 @@ flutter:
         - asset: fonts/custom-icons.ttf
   assets:
     - assets/i18n/
+flutter_localizations: # add this
+    sdk: flutter
```
Comment (validcube) [inline pubspec.yaml:106]: Two spaces, not four. grrr

```suggestion
  sdk: flutter
```

File: pubspec.yaml [hunk 4]
```diff
@@ -81,6 +80,8 @@ dependencies:
   flutter_file_dialog: ^3.0.2
   wakelock_plus: ^1.1.3
   share_plus: ^7.2.1
+  slang: ^3.25.0
+  slang_flutter: ^3.25.0
 
 dev_dependencies:
```
Comment (validcube) [inline pubspec.yaml:86]: Missing `slang_build_runner` as dev dependency

https://pub.dev/packages/slang#getting-started

**Review by validcube — CHANGES_REQUESTED** [review]
Bleep

File: pubspec.yaml (hunk 4 above)
Comment (validcube) [reply pubspec.yaml:86]: Fixed: 11086213276917d9ccefc80259d444460f264485

**Comment by Aunali321** [comment]
image

**Review by Aunali321 — APPROVED** [review]
image

**Review by validcube — CHANGES_REQUESTED** [review]
Bloop: The documentation should be updated to mentions that translation must be generated using the command:

```bash
dart run slang
```

because we are not using [slang build runner](https://pub.dev/packages/slang_build_runner)

----------------------------------------------------------

👀 Checkout the full conversation discussed @ `Discord/ReVanced (Team chat; private)`
https://canary.discord.com/channels/952946952348270622/952987428786941952/1168894670579966012

**Review by validcube — APPROVED** [review]
🥞 Tested & Approved

**Comment by oSumAtrIX** [comment]
Are the translation strings supposed to be included in this PR? They should be added separately by Crowdin

**Comment by validcube** [comment]
> @oSumAtrIX Are the translation strings supposed to be included in this PR? They should be added separately by Crowdin

I'm sure that they aren't supposed to be included because it's out-of-scope, I decided to add translation to this PR because the repository don't have translation merged yet and I don't really know how would I (automatically) add translation to the repository.

On top of that, the translation file have to be renamed from *.json to strings_*.i18n.json (strings_ are optional).

**Comment by Ushie** [comment]
That change can be done in Crowdin, once this PR is ready to merge

**Comment by validcube** [comment]
> @Ushie That change can be done in Crowdin, once this PR is ready to merge

This PR is ready to be merge into `dev`.

> @validcube I don't really know how would I (automatically) add translation to the repository because the translation file have to be renamed from *.json to strings_*.i18n.json

On top of that, the translation need to be clean up due to missing translation in base language

**Comment by oSumAtrIX** [comment]
Alright, in this case, you can remove the translations via a commit which will be squashed with the rest of the commits

**Comment by Ushie** [comment]
- [ ] Please work on the translation popup, the lack of actions makes it look odd so you can add an "OK" button or such
- [ ] Is there any way to use friendly language names? e.g. English (US)

**Comment by oSumAtrIX** [comment]
@Ushie Can you assign the person who you refer to?

**Comment by validcube** [comment]
> Please work on the translation popup, the lack of actions makes it look odd so you can add an "OK" button or such
> Is there any way to use friendly language names? e.g. English (US)

🥞 All of these have been fixed in 3c18c4e.

**Review by Aunali321 — APPROVED** [review]
Nice work!, looks good to me

**Review by validcube — APPROVED** [review]
🥞 Approved :troll, self-approved

**Review by validcube — COMMENTED** [review]
The only thing that left is to resolve conflicts and rebase changes on of latest dev branch

**Pull request merged**